### PR TITLE
feat: invoice notification

### PR DESCRIPTION
**Pull request by aguirrekarina**
Implement sending an email after making a purchase. 
For this functionality, the Stripe web-hook must be activated.

Command to use:

```bash
stripe listen --forward-to localhost:5001/api/payment/Stripe/web-hook/register-web-hook-event
```

Code reviewes:
@daniel-espinoza-79 
@JeferssonCL

### CODE REVIEW

**Review by daniel-espinoza-79 — APPROVED** [review]
Good Job

**Review by JeferssonCL — CHANGES_REQUESTED** [review]
I think that you can include information about roduct details such as variants and attributes because we are modifying price based on it

File: apps/merchant-api/PaymentService/src/PaymentService.Application/QueryCommands/StripeWebHookRegister/Commands/CommandHandlers/CreateEventRegisterWebHookCommandHandler.cs [hunk 1]
```diff
@@ -115,10 +127,77 @@ private async Task<BaseResponse> ProcessPaymentTransaction(Session session, Succ
...
+                orderResponse.Data.ToString(), 
+                orderItems,
+                (decimal)session.AmountTotal! / 100m
+            )
+        );
+
+        await _producer.Publish(orderEmail);
+    }
+    
+    private async Task<BaseResponse> GetProductVariantDetails(Guid productVariantId)
```
Comment (JeferssonCL) [inline apps/merchant-api/PaymentService/src/PaymentService.Application/QueryCommands/StripeWebHookRegister/Commands/CommandHandlers/CreateEventRegisterWebHookCommandHandler.cs]: This part of code is a comunnication with another service as a client so is not correct be here soyou can move this part or segment of code to product client service.

Also I'm having this problem, do you know how can I fix it the email is not arriving to me: 
Captura desde 2024-12-05 10-54-27

**Review by JeferssonCL — CHANGES_REQUESTED** [review]

**Review by JeferssonCL — APPROVED** [review]
Good job

**Pull request merged**